### PR TITLE
Tests: Add more tests for `JsonArray`

### PR DESCRIPTION
**Pull request by kemzeb**
At an attempt to detect future regressions in AK/JsonArray, this snapshot adds additional tests for it to TestJSON.cpp.

See https://github.com/SerenityOS/serenity/issues/18618 for more context.

### CODE REVIEW

**Comment by kemzeb** [comment]
The most recent snapshot change is due to a commit I removed since it does not relate to his PR.

File: Tests/AK/TestJSON.cpp [hunk 1]
```diff
@@ -374,6 +374,143 @@ TEST_CASE(fallible_json_array_for_each)
...
+    EXPECT(array.is_empty());
+
+    raw_json = "[1, 2]"sv;
+    json_value = MUST(JsonValue::from_string(raw_json));
+    array = json_value.as_array();
+    EXPECT(!array.is_empty());
+}
+
+// Declaration exists to silence -Werror=missing-declarations.
+JsonArray setup_json_array();
```
Comment (Hendiadyoin1) [inline Tests/AK/TestJSON.cpp:391]: you could mark it `static` instead to avoid this issue as well

**Review by AtkinsSJ — APPROVED** [review]
Thanks!

**Comment by kemzeb** [comment]
Of course!

**Pull request merged**